### PR TITLE
Adding walking entity.

### LINKED ISSUE #72: Add `Walking` entity.
## This task is a part of the issue:
 - #65 

### Dev requirements and restrictions:
```swift
struct Walking {
   id: UUID
   sitterID: UUID
   ownerID: UUID?
   Status: Int or String
   start: DateStamp
   end: DateStamp
   PlaceOfWalking: GeoLocation? 
   Notes: String? 
   Review: String
   Rating: Double
}
```

Statuses for walks:
- Available - when the walk is created.
- Pending - awaiting confirmation from the sitter.
- Booked - when the sitter confirms the walk.
- Finished - after the completion of the walk.
- CancelledByOwner - the owner canceled the walk.
- CancelledBySitter - the sitter canceled the walk.
- Expired - the walk was not booked, and the date and time have already passed.

### PR DESCRIPTION
**Pull request by iBYarchitect**
## Summary
This PR is a part of the work to resolve #72 

## Changes
- Added struct `Walking`
- Added struct `Ratings`
- Added struct `Location`
- Added struct `Reviews`
- Added enum `WalkingStatus`
- Added enum `Dummy` with information for Walking instances 

## Checks to complete
- [x] The PR title has a conventional commit style (short and descriptive).
- [x] The PR description includes essential information that helps understand the purpose of the changes.
- [x] I've built and run my changes, and no warnings or errors occurred.
- [x] All existing tests passed with my changes.
- [x] My code follows our style guide.
- [x] I've performed a self-review of my code.
- [x] Every non-private interface is documented properly.
- [x] I've added tests for my code (if it makes sense).
- [x] I've updated `CHANGELOG.md` file.
- [x] PR has assignee, reviewers, milestone, it is properly linked to the project and to the issue.

### CODE REVIEW

File: Woof/Woof/Model/Walking/Location.swift [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import Foundation
+
+/**
+ The geographical location of the walking service.
+ Conforms to the `Codable` protocols.
```
Comment (kirpichonok) [inline Woof/Woof/Model/Walking/Location.swift:5]: Why do we need it? I can see in code which protocols the type adopts.

File: Woof/Woof/Model/Walking/Ratings.swift [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import Foundation
+
+/**
+ The ratings given by both the pet owner and the pet sitter
+ Conforms to the `Codable` protocols.
+ */
+struct Ratings: Codable {
```
Comment (kirpichonok) [inline Woof/Woof/Model/Walking/Ratings.swift:7]:
```suggestion
struct Rating: Codable {
```
We seemed discussed that we should use singular.

File: Woof/Woof/Model/Walking/Ratings.swift [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+    /// The rating is given to walking by the pet owner.
+    let ownerRating: Int
```
Comment (kirpichonok) [inline Woof/Woof/Model/Walking/Ratings.swift:9]: Can the user set a rating with "1000" value? How that will be tracked off? We've already had the `StarRating`. Why not use a similar solution?

File: Woof/Woof/Model/Walking/Location.swift [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import Foundation
```
Comment (kirpichonok) [inline Woof/Woof/Model/Walking/Location.swift:1]: It's a superfluous line.

File: Woof/Woof/Model/Walking/WalkingStatus.swift [hunk 5]
```diff
@@ -0,0 +1,43 @@
+import Foundation
+
+/**
+ The possible walking status values for a `Walking`.
```
Comment (kirpichonok) [inline Woof/Woof/Model/Walking/WalkingStatus.swift:4]: nit:
```suggestion
The current walking status.
```

File: Woof/Woof/Model/Walking/WalkingStatus.swift [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+ Conforms to the `Codable` protocol.
+ */
+enum WalkingStatus: Codable {
+    /// The walking is available for booking.
+    case available
+
+    /// The walking is pending approval from the pet sitter.
+    case pending
+
+    /// The walking is booked by owner.
```
Comment (kirpichonok) [inline Woof/Woof/Model/Walking/WalkingStatus.swift:14]: I think it's more correct and consistent:
```suggestion
    /// The walking has been booked by owner.
```

File: Woof/Woof/Model/Walking/WalkingStatus.swift [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+    case finished
+
+    /// The walking has been canceled by the sitter.
+    case canceledBySitter
+
+    /// The walking has expired due to the elapsed duration since it was initially declared.
+    case expired
+
+    /// Text description of possible status.
+    var description: String {
```
Comment (kirpichonok) [inline Woof/Woof/Model/Walking/WalkingStatus.swift:27]: Why is this property needed? Maybe make the _enum_ with _String_ associated type: 
`enum WalkingStatus: String, Codable { ..}` 
and use _rawValue_?

**Review by kirpichonok — COMMENTED** [review]
Great job as always! 
But I have some questions as always 😁
If you agree with my comments, see the rest of the code.
And it's difficult to understand the purpose of the PR without usage of the new types.

File: Woof/Woof/Model/Walking/Reviews.swift [hunk 8]
```diff
@@ -0,0 +1,13 @@
+import Foundation
+
+/**
+ The owner and sitter reviews for the walking service.
+ Conforms to the `Codable` protocol.
+ */
```
Comment (NadzeyaShpakouskaya) [inline Woof/Woof/Model/Walking/Reviews.swift:6]: Nit: I think we can not specify whose reviews is in structure.
```suggestion
/**
 The reviews for the walking service.
 */
```

File: Woof/Woof/Model/Walking/Walking+Dummy.swift [hunk 9]
```diff
@@ -0,0 +1,75 @@
+import Foundation
+
+extension Walking {
+    /// Provides dummy data for `Walking` model.
+    enum Dummy {
+        static let dummyWalking = Walking(
```
Comment (NadzeyaShpakouskaya) [inline Woof/Woof/Model/Walking/Walking+Dummy.swift:6]: Missing doc

File: Woof/Woof/Model/Walking/Walking+Dummy.swift [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+            start: Date(),
+            end: Date() + 2400,
+            location: Location(longitude: 0.0, latitude: 0.0, address: "Sample Address"),
+            reviews: Reviews(ownerReview: "", sitterReview: ""),
+            rating: Ratings(ownerRating: 0, sitterRating: 0),
+            notes: "Sample notes",
+            price: 28.0 // You can set the price as needed
+        )
+
+        static let bulkDummyWalkings = [
```
Comment (NadzeyaShpakouskaya) [inline Woof/Woof/Model/Walking/Walking+Dummy.swift:20]: Missing doc

File: Woof/Woof/Model/Walking/Walking+Dummy.swift [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+            Walking(
+                id: UUID(),
+                owner: nil,
+                sitter: Sitter(), // You may want to create a Sitter instance here
+                status: .finished, // You can change the status as needed
```
Comment (NadzeyaShpakouskaya) [inline Woof/Woof/Model/Walking/Walking+Dummy.swift:25]: Do we need these comments?
Please review the rest of the code.

File: Woof/WoofTests/Model/Walking/WalkingStatusTests.swift [hunk 12]
```diff
@@ -0,0 +1,13 @@
+import XCTest
+
+final class WalkingStatusTests: XCTestCase {
+    func testDescriptionReturnsExpectedStringAndExist() {
```
Comment (NadzeyaShpakouskaya) [inline Woof/WoofTests/Model/Walking/WalkingStatusTests.swift:4]: Can you add the test to check all statuses are presented?

**Review by NadzeyaShpakouskaya — CHANGES_REQUESTED** [review]
Nice attempt, could you please add unit tests for all created models, at least that they can be initialized/called?
Also, please consider some minor improvements:

File: Woof/Woof/Model/Walking/Ratings.swift (hunk 3 above)
Comment (iBYarchitect) [reply Woof/Woof/Model/Walking/Ratings.swift]: I'll change the type to Double.
In my opinion, implementing rating validation is the responsibility of the ViewModel.

File: Woof/Woof/Model/Walking/Ratings.swift (hunk 2 above)
Comment (iBYarchitect) [reply Woof/Woof/Model/Walking/Ratings.swift]: Fixed with 7f70e76

File: Woof/Woof/Model/Walking/Location.swift (hunk 1 above)
Comment (iBYarchitect) [reply Woof/Woof/Model/Walking/Location.swift]: You're right, just a few lines later you can see this in the class declaration.
Deleted with 7f70e76

File: Woof/Woof/Model/Walking/Location.swift (hunk 4 above)
Comment (iBYarchitect) [reply Woof/Woof/Model/Walking/Location.swift]: Fixed with 7f70e76

File: Woof/Woof/Model/Walking/WalkingStatus.swift (hunk 5 above)
Comment (iBYarchitect) [reply Woof/Woof/Model/Walking/WalkingStatus.swift:2]: I guess that enum describes all possible variants, not a current value of walking status.

File: Woof/Woof/Model/Walking/WalkingStatus.swift (hunk 7 above)
Comment (iBYarchitect) [reply Woof/Woof/Model/Walking/WalkingStatus.swift]: Yes, the associated type it's a more lightweight variant.
Fixed with 7f70e76

File: CHANGELOG.md [hunk 13]
```diff
@@ -6,6 +6,10 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.1.0/).
 
 ## [Unreleased]
 
+### Added 
+
+- Added `Walking` enity - [#88](https://github.com/IronFoundation/iOS-Woof/pull/88)
+
```
[the remark this answers is not in the document]
Comment (iBYarchitect) [reply CHANGELOG.md]: Fixed with 7f70e76

File: Woof/Woof/Model/Walking/Location.swift (hunk 1 above)
[the remark this answers is not in the document]
Comment (iBYarchitect) [reply Woof/Woof/Model/Walking/Location.swift]: You're right, fixed with 7f70e76

File: Woof/Woof/Model/Walking/Ratings.swift [hunk 14]
```diff
@@ -0,0 +1,13 @@
+import Foundation
+
+/**
+ The ratings given by both the pet owner and the pet sitter
+ Conforms to the `Codable` protocols.
```
[the remark this answers is not in the document]
Comment (iBYarchitect) [reply Woof/Woof/Model/Walking/Ratings.swift]: You're right, fixed with 7f70e76

File: Woof/Woof/Model/Walking/Reviews.swift (hunk 8 above)
Comment (iBYarchitect) [reply Woof/Woof/Model/Walking/Reviews.swift]: Fixed with 7f70e76

File: Woof/Woof/Model/Walking/Walking+Dummy.swift (hunk 9 above)
Comment (iBYarchitect) [reply Woof/Woof/Model/Walking/Walking+Dummy.swift]: In my opinion, you could understand the purpose of these constants from their names.
Fixed with 7f70e76

File: Woof/Woof/Model/Walking/Walking+Dummy.swift (hunk 10 above)
Comment (iBYarchitect) [reply Woof/Woof/Model/Walking/Walking+Dummy.swift:22]: In my opinion, you could understand the purpose of these constants from their names.
Fixed with 7f70e76

File: Woof/Woof/Model/Walking/Walking+Dummy.swift (hunk 11 above)
Comment (iBYarchitect) [reply Woof/Woof/Model/Walking/Walking+Dummy.swift]: Perhaps they can be useful for a person who wants to change something.
Fixed with 7f70e76

File: Woof/WoofTests/Model/Walking/WalkingStatusTests.swift (hunk 12 above)
Comment (iBYarchitect) [reply Woof/WoofTests/Model/Walking/WalkingStatusTests.swift:4]: Added with 3239009

File: Woof/WoofTests/Model/Walking/WalkingTests.swift [hunk 15]
```diff
@@ -0,0 +1,24 @@
+import XCTest
+
+final class WalkingTests: XCTestCase {
+    func testWalInitialization() {
```
Comment (NadzeyaShpakouskaya) [inline Woof/WoofTests/Model/Walking/WalkingTests.swift:4]: I don't get the name of test :)

File: Woof/Woof/Model/Walking/Ratings.swift (hunk 3 above)
Comment (NadzeyaShpakouskaya) [reply Woof/Woof/Model/Walking/Ratings.swift]: @iBYarchitect Why do you change it to Double? Will users rate it as 4.5, when they write a review? I think the `Int` is ok for this case.

**Review by NadzeyaShpakouskaya — APPROVED** [review]
Overall LGTM

**Pull request merged**